website/blog/2022-12-13-using-cpts-and-taxonomies.md

```diff
@@ -0,0 +1,61 @@
+
+Post `category` and `tag` are taxonomies - a grouping you can further separate by terms belonging to that taxonomy. WordPress, since version 2.3.0 offers a way to register your own, custom taxonomies.
+
+## Registering Custom Post Types
+For this exercise, we'll create a new post type called `Projects`. To make the process of registering new CPTs as easy as possible, we'll use a WP-CLI command to create our CPT with the following command:
+
+```bash
+wp boilerplate create post-type --label='Project' --plural_label='Projects' --slug='project' --rewrite_url='project' --rest_endpoint_slug='projects'
+```
+
+> If your new CPT is not working, try flushing rewrite rules by re-saving the settings in **Settings -> Permalinks** or by using `wp cache flush` CLI command
+
+Your new post type is registered and ready to use! Easy, right?
+
+Sometimes all these parameters can be a bit confusing, so here's a quick reference of best practices when setting these parameters:
+
+| Parameter          | Singular/Plural | Writing style | Example  |
+|--------------------|-----------------|---------------|----------|
+| label              | Singular        | Regular       | Project  |
+| plural_label       | Plural          | Regular       | Projects |
+| slug               | Singular        | kebab-case    | project  |
+| rewrite_url        | Singular        | kebab-case    | project  |
+| rest_endpoint_slug | Plural          | kebab-case    | projects |
+
+## Registering Taxonomies
+Now that we have the new custom post type, we need a way to group the projects. We'll create a custom taxonomy called `Project Category`. As with the CPT registration, the easiest way to register taxonomies is by using the following WP-CLI command:
+
+```bash
+wp boilerplate create taxonomy --label='Project Category' --plural_label='Project Categories' --slug='project-category' --rest_endpoint_slug='project-categories' --post_type_slug='project'
+```
+
+Similar suggestions apply to the parameters when naming taxonomies as well. Be sure to write the correct post type slug for which you are registering this new taxonomy!
+
+## Structure
+When checking your codebase after adding these new custom post types and taxonomies, you'll notice the post types are located inside the `src/CustomPostType` folder, and the taxonomies are located inside the `src/CustomTaxonomy` folder. Following the **Single Responsibility Principle**, each post type or taxonomy is in a separate class.
+
+## Modifying options
+Our custom post type and taxonomy are ready to use, but we still want to make some changes. For starters, we want another icon and for the Projects to be located below the Posts in the admin menu. In `src/CustomPostType/ProjectPostType.php`, find the `MENU_POSITION` constant and change it to `5`. The lower the number, the higher it will be in the menu.
```

Comment:
```suggestion
Our custom post type and taxonomy are ready to use, but we still want to make some changes. For starters, we want another icon and for the Projects to be located below the Posts in WordPress admin menu. In `src/CustomPostType/ProjectPostType.php`, find the `MENU_POSITION` constant and change it to `5`. The lower the number, the higher it will be in the menu.
```

Reply:
As a first reference to the menu it might be good to say explicitly which menu it is :)